tests/conftest.py

```diff
@@ -162,6 +162,11 @@ def reset_integrations():
 @pytest.fixture
 def sentry_init(request):
     def inner(*a, **kw):
+        # Clear scopes to not leak scope data between tests
+        sentry_sdk.scope._global_scope = None
```

Comment:
this is also a code smell, we did not need to do this cleanup before. 
imo you should implement the right logic in `sentry_sdk.init` and setup intial state properly and not here in the tests.

Reply:
Hmm how would this work?
Would we reset global scope on `Sentry.init`? Doesn't this throw away all data that was set on the global scope before `Sentry.init`?
Also what should happen if `Sentry.init` is executed multiple times? Can we simply keep global scope as is?

Reply:
It kind of is. The test setup is still Hub based and when we run the tests there are a lot of `init()` calls in one process and all kind of code that manipulates the scope in various ways. It is kind of not how we intend the sdk to be used. This is also why we have lots of `@pytest.mark.forked` in our tests. 
I would leave this in there now and when we upgrade our integrations to scopes also upgrade the test setup to scopes and revisit this then.

Reply:
I am also with @adinauer in the RFC it is also mentioned that it should be possible to mutate the global scope before `init()` is called: https://github.com/getsentry/rfcs/blob/fn/merge-hub-scope/text/0122-sdk-hub-scope-merge.md?plain=1#L219